src/libraries/Common/src/System/Data/ProviderBase/DbMetaDataFactory.cs

```diff
@@ -278,7 +278,7 @@ internal DataRow FindMetaDataCollectionRow(string collectionName)
                             exactCollectionName = candidateCollectionName;
                             haveExactMatch = true;
                         }
-                        else
+                        else if (haveExactMatch == false)
```

Comment:
```suggestion
                        else if (!haveExactMatch)
```

Reply:
Should this be changed? The file already use `== false` everywhere else.

Reply:
I've always considered boolean comparisons within an if condition to be redundant. It's like saying if (true == true) == true. But I'm fine leaving it as-is to follow existing precedent in the file.